core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -275,17 +275,23 @@ object SparkSubmit {
 
     // Resolve maven dependencies if there are any and add classpath to jars. Add them to py-files
     // too for packages that include Python code
+    val exclusions: Seq[String] =
+      if (!Utils.isEmptyString(args.packagesExclusions)) {
+        args.packagesExclusions.split(",")
+      } else {
+        Nil
+      }
     val resolvedMavenCoordinates =
       SparkSubmitUtils.resolveMavenCoordinates(
-        args.packages, Option(args.repositories), Option(args.ivyRepoPath))
-    if (!resolvedMavenCoordinates.trim.isEmpty) {
-      if (args.jars == null || args.jars.trim.isEmpty) {
+        args.packages, Option(args.repositories), Option(args.ivyRepoPath), exclusions)
```

Comment:
super nit: since `exclusions` has a default value, we generally use the longer syntax:

    resolvedMavenCoordinates(blah, blah, exclusions = exclusions)